### PR TITLE
Update PTQ docs

### LINKED ISSUE #2540: POT -> PTQ transition in docs
Steps to be done:
- Rename POT -> PTQ
- Replace POT docs links with NNCF.PTQ docs
- Mention that the output model would be quantized to INT8

Related pages: 
https://openvinotoolkit.github.io/training_extensions/stable/guide/explanation/additional_features/models_optimization.html
https://openvinotoolkit.github.io/training_extensions/stable/guide/tutorials/base/how_to_train/classification.html#optimization and other tasks

### PR DESCRIPTION
**Pull request by GalyaZalesskaya**
### Summary

Update in the documentation:
  - Rename POT -> PTQ
  - Replace POT docs links with NNCF.PTQ docs
  - Mention that the output model would be quantized to INT8

[Issue#2540](https://github.com/openvinotoolkit/training_extensions/issues/2540)

resolves #2540

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: docs/source/guide/explanation/additional_features/models_optimization.rst [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+OpenVINO™ Training Extensions provides two types of optimization algorithms: `Post-Training Quantization tool (PTQ) <https://github.com/openvinotoolkit/nncf#post-training-quantization>`_ and `Neural Network Compression Framework (NNCF) <https://github.com/openvinotoolkit/nncf>`_.
 
 *******************************
-Post-training Optimization Tool
+Post-Training Quantization Tool
 *******************************
 
-POT is designed to optimize the inference of models by applying post-training methods that do not require model retraining or fine-tuning. If you want to know more details about how POT works and to be more familiar with model optimization methods, please refer to `documentation <https://docs.openvino.ai/latest/pot_introduction.html#doxid-pot-introduction>`_.
+PTQ is designed to optimize the inference of models by applying post-training methods that do not require model retraining or fine-tuning. If you want to know more details about how PTQ works and to be more familiar with model optimization methods, please refer to `documentation <https://docs.openvino.ai/latest/pot_introduction.html#doxid-pot-introduction>`_.
```
Comment (sovrasov) [inline docs/source/guide/explanation/additional_features/models_optimization.rst]: This link is more relevant than description of deprecated POT https://docs.openvino.ai/2023.2/ptq_introduction.html

File: docs/source/guide/explanation/additional_features/models_optimization.rst [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
 
-To run Post-training optimization it is required to convert the model to OpenVINO™ intermediate representation (IR) first. To perform fast and accurate quantization we use ``DefaultQuantization Algorithm`` for each task. Please, see the `DefaultQuantization Parameters <https://docs.openvino.ai/latest/pot_compression_algorithms_quantization_default_README.html#doxid-pot-compression-algorithms-quantization-default-r-e-a-d-m-e>`_ for further information about configuring the optimization.
+To run Post-training quantization it is required to convert the model to OpenVINO™ intermediate representation (IR) first. To perform fast and accurate quantization we use ``DefaultQuantization Algorithm`` for each task. Please, see the `DefaultQuantization Parameters <https://docs.openvino.ai/latest/pot_compression_algorithms_quantization_default_README.html#doxid-pot-compression-algorithms-quantization-default-r-e-a-d-m-e>`_ for further information about configuring the optimization.
```
Comment (sovrasov) [inline docs/source/guide/explanation/additional_features/models_optimization.rst]: DefaultQuantization Algorithm doesn't exist anymore, we may just reference
Tune quantization parameters https://docs.openvino.ai/2023.2/basic_quantization_flow.html#tune-quantization-parameters here

File: docs/source/guide/get_started/cli_commands.rst [hunk 3]
```diff
@@ -419,7 +419,7 @@ With the ``--help`` command, you can list additional information, such as its pa
       --test-data-roots TEST_DATA_ROOTS
                             Comma-separated paths to test data folders.
       --load-weights LOAD_WEIGHTS
-                            Load model weights from previously saved checkpoint.It could be a trained/optimized model (POT only) or exported model.
+                            Load model weights from previously saved checkpoint. It could be a trained/optimized model (with PTQ only) or exported model.
```
Comment (sovrasov) [inline docs/source/guide/get_started/cli_commands.rst:422]: Could you update code of CLI as well? Looks like it still references POT

File: docs/source/guide/tutorials/advanced/api_tutorial.rst [hunk 4]
```diff
@@ -181,7 +181,7 @@ To validate the OpenVINO™ IR model, we need to create an openvino task first a
 Optimization
 ************
 
-To run the optimization with POT on the OpenVINO™ IR model, we need to create an output model and run the optimization procedure:
+To run the optimization with PTQ (also known as POT) on the OpenVINO™ IR model, we need to create an output model and run the optimization procedure:
```
Comment (sovrasov) [inline docs/source/guide/tutorials/advanced/api_tutorial.rst]: POT is completely different from PTQ, so let's omit it

**Review by sovrasov — COMMENTED** [review]
Thanks for the PR, this makes our docs much more consistent. Could you also merge the latest changes from develop?

**Comment by GalyaZalesskaya** [comment]
@sovrasov Thank you for your feedback, applied your comments

**Review by sovrasov — APPROVED** [review]

**Review by wonjuleee — APPROVED** [review]

**Pull request merged**